src/frontend/ui/ui_manager.rs

```diff
@@ -45,21 +52,21 @@ pub struct UiManager {
     canvas: Canvas,
     selection: Rectangle,
     drawables: Vec<Box<dyn Drawable>>,
-    app_model_sender: Sender<Command>,
+    sender: ComponentSender<AppModel>,
 
     #[derive_where(skip(Debug))]
     render_observer: Vec<Box<RenderObserver>>,
 }
 
 impl UiManager {
-    pub fn new(total_width: i32, total_height: i32, app_model_sender: Sender<Command>) -> Self {
+    pub fn new(total_width: i32, total_height: i32, sender: ComponentSender<AppModel>) -> Self {
```

Comment:
The reason why I renamed it to `sender` is that I think its type makes it clear that it's a sender from `AppModel`.